### DISCUSSION TITLE
Problems with nvidia driver on 12

### DISCUSSION THREAD

**Discussion by geoubuntu** [SpiralLinux/SpiralLinux-project#317] [Q&A]
Hi. With 6.4 kernel I get an error installing nvidia driver from debian repos. So ai tried to disinstall kernel 6.4 and use only 6.1 but synaptic is unable to erase the kernel 6.4 ( while using 6.1) . Can you help me with both of the problems, install nvidia on 6.4 and delete 6.4 and use only 6.1? Thanks

**Comment by geckolinux** [comment]
Hi there, what's the error?

**Comment by geoubuntu** [comment]
I installed debian 12 instead. Ai remember the error was in dkms on
building on linux 6.4 . Can you teach me how to disintall 6.4?

**Reply by geckolinux** [reply]
Hi there, in order to help we would need to know what errors you saw in SpiralLinux.

**Comment by fweller** [comment]
Hello @geoubuntu 

Supporting nvidia GPU works slightly differently between Ubuntu and Debian, I am assuming you previously used Ubuntu based on your username.  I promised myself that I would never purchase an nvidia GPU ever again, but my employer thought differently.  Below are my notes on how I install nvidia drivers in Debian.  These instructions work for me in debian 12 bookworm.  Your request to switch kernel versions (install the version you want, reboot, uninstall the version you don't want) can be handled in the synaptic package manager.

# Remove all nvidia stuff from a system
su
apt list -i | grep nvidia
dpkg -l | grep nvidia
apt purge nvidia-driver-*
apt purge nvidia-*

# Confirm GPU exists
su
lspci -nn | egrep -i "3d|display|vga"
lspci | grep -e VGA

# Install nvidia stuff
su
apt update
apt install linux-headers-amd64   # Linux headers
apt install firmware-misc-nonfree # All firmware
apt install nvidia-driver         # nvidia driver
apt install nvidia-xconfig        # The NVIDIA X Configuration Tool
apt install nvidia-smi            # NVIDIA System Management Interface
nvidia-settings   # Save X config file to /etc/X11/xorg.conf
# a message may appear, this is normal:
# The free nouveau kernel module is currently loaded and conflicts with the non-free nvidia kernel module.
Power off the computer, power off any attached hubs or docks, then turn power back on


# Debugging
su
lshw -c video         # List hardare
lsmod | grep nvidia   # List nvidia drivers
lspci | grep VGA      # List all video hardware
lspci -vnnn | perl -lne 'print if /^\d+\:.+(\[\S+\:\S+\])/' | grep VGA  # Show nvidia on PCIe
lspci | grep ' VGA ' | cut -d" " -f 1 | xargs -i lspci -v -s {}         # Show nvidia on PCIe details
lsof /dev/nvidia*     # Is nvidia active?
inxi -G               # Graphics system information
xrandr                # List vido capabilities for all attached displays
xrandr --verbose      # Do it verbosely
xrandr --listproviders  # List GPUs
xrandr --listmonitors   # List Displays that are active
nvidia-xconfig --query-gpu-info   # Get info that nvidia knows about

**Reply by geckolinux** [reply]
Thanks very much for the excellent instructions. Since @geoubuntu mentioned something about DKMS I assume it was this issue, now fixed: https://github.com/SpiralLinux/SpiralLinux-project/issues/315
